packages/harpguru-core/src/components/harp-cell/hooks/use-set-pozition-root/use-set-pozition-root.ts

```diff
@@ -21,11 +26,25 @@ export const useSetPozitionRoot = (): SetPozitionRoot => {
       rootPitchId: newRootPitchId,
     }
     const { pozitionId: newPozitionId } = getCovariantSet(covariantControllers)
-    const newActiveHarpStrata = getHarpStrata({
-      ...activeHarpStrataProps,
-      harpKeyId: activeHarpKeyId,
-      pozitionId: newPozitionId,
-    })
-    setActiveHarpStrata(newActiveHarpStrata)
+    if (lockedCovariant === CovariantMembers.HarpKey) {
+      const newActiveHarpStrata = getHarpStrata({
+        ...activeHarpStrataProps,
+        harpKeyId: activeHarpKeyId,
+        pozitionId: newPozitionId,
+      })
+      setActiveHarpStrata(newActiveHarpStrata)
+    } else {
+      const covariantControllers2 = {
+        rootPitchId: activeRootPitchId,
+        pozitionId: newPozitionId,
+      }
+      const { harpKeyId: newHarpKeyId } = getCovariantSet(covariantControllers2)
+      const newActiveHarpStrata = getHarpStrata({
+        ...activeHarpStrataProps,
+        harpKeyId: newHarpKeyId,
+        pozitionId: newPozitionId,
+      })
+      setActiveHarpStrata(newActiveHarpStrata)
+    }
```

Comment:
I wonder whether this should be shouting the exact shift which has happened out to the user rather than just making the switch..

Reply:
#22 raised.